core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -73,9 +73,13 @@ class NettyBlockTransferService(conf: SparkConf, securityManager: SecurityManage
       val server = transportContext.createServer(port, bootstraps.asJava)
       (server, server.getPort)
     }
-
     val portToTry = conf.getInt("spark.blockManager.port", 0)
-    Utils.startServiceOnPort(portToTry, startService, conf, getClass.getName)._1
+
+    if(Utils.hasPortRange(conf)){
+      Utils.startServiceFromPortRange(portToTry, startService, conf, getClass.getName)._1
```

Comment:
Maybe I am missing something but why can't we do this

def startServiceOnPort(....) = {
   if(Utils.hasPortRange(conf)) {
      startServiceFromPortRange(...)
  } else {
     //existing code
 }
}

Then we don't have to change any Netty* or AkkaUtils. This will keep the logic closer together as well.